recipe-client-addon/test/browser/browser_FilterExpressions.js

```diff
@@ -47,8 +61,32 @@ add_task(async function() {
   is(val, true, "Bucket sample returns true for a known sample");
   val = await FilterExpressions.eval('["test-4"]|bucketSample(0, 5, 10)');
   is(val, false, "Bucket sample returns false for a known sample");
+});
 
-  // Test that it reads from the context correctly.
-  val = await FilterExpressions.eval("first + second + 3", {first: 1, second: 2});
-  is(val, 6, "context is available to filter expressions");
+// Preference tests
+add_task(async function() {
+  let val;
+  // Compare the value of the preference
+  await SpecialPowers.pushPrefEnv({set: [["normandy.test.value", 3]]});
+  val = await FilterExpressions.eval('"normandy.test.value"|preferenceValue == 3');
+  is(val, true, "preferenceValue expression compares against preference values");
+
+  // preferenceValue can take a default value as an optional argument, which
+  // defaults to `undefined`.
+  val = await FilterExpressions.eval('"normandy.test.default"|preferenceValue(false) == false');
+  is(val, true, `preferenceValue takes optional 'default value' param for prefs without set values`);
+
+  val = await FilterExpressions.eval('"normandy.test.value"|preferenceValue(5) == 3');
+  is(val, true, `preferenceValue default param is not returned for prefs with set values`);
+
+  // Compare if the preference is user set
+  val = await FilterExpressions.eval('"normandy.test.isSet"|preferenceIsUserSet != true');
+  is(val, true, "preferenceIsUserSet expression determines if preference is even set");
+
+  val = await FilterExpressions.eval('"normandy.test.value"|preferenceIsUserSet == true');
+  is(val, true, "preferenceIsUserSet expression determines if user set preference themselves");
+
+  // Compare if the preference has _any_ value, whether it's user-set or default,
+  val = await FilterExpressions.eval('"normandy.test.nonexistant"|preferenceExists == false');
+  is(val, true, "preferenceExists expression determines if preference exists at all");
```

Comment:
Every single one of the filter expressions in this task succeeds/passes/returns true. We should have corresponding tests for the cases where they don't.